python/hawkey/iutil-py.cpp

```diff
@@ -127,6 +130,34 @@ advisoryRefVectorToPylist(const std::vector<libdnf::AdvisoryRef> & advisoryRefs,
     return list.release();
 }
 
+PyObject *
+changelogslist_to_pylist(const GPtrArray *changelogslist)
+{
+    UniquePtrPyObject list(PyList_New(0));
+    if (!list)
+        return NULL;
+    PyDateTime_IMPORT;
+
+    for (unsigned int i = 0; i < changelogslist->len; ++i) {
+        auto citem = static_cast<libdnf::Changelog *>(g_ptr_array_index(changelogslist, i));
+        PyObject *d = PyDict_New();
+        if (!d)
+            return NULL;
+        if (PyDict_SetItemString(d, "author", PyUnicode_FromString(citem->author)) == -1)
+            return NULL;
```

Comment:
Who frees dictionary `d`?
The same problem is repeating. What about use `UniquePtrPyObject` as reference guard?